### PR TITLE
Add Unit Review questions to Sprint Review form

### PR DESCRIPTION
**Pull request by mellogarrett**
## Jira Issue

[CORE-1605]

## Description

- Conditionally add the Unit Review questions to the end of the Sprint Review form based on if the TL selects a Sprint Submission that has a corresponding Unit Submission or not.
- Update Sprint Retro & Review API code to the updated `atlas-db` interface.
- Remove standalone Unit Retro & Review forms.
- Update `question-key` types to accommodate iterating over the key sets.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

[CORE-1605]: https://lambdaschool.atlassian.net/browse/CORE-1605

### CODE REVIEW

**Comment by mellogarrett** [comment]
This build is failing because it needs the new atlas-db version.

File: forms/sprint-review.ts [hunk 1]
```diff
@@ -133,5 +133,91 @@ Student can access this feedback through their dashboard`,
...
+          range: { start: 1, end: 3 },
+          title: 'Unit Assessment Score 6',
+          isRequired: false,
+          key: 'unitRubricItem6Rating',
+        },
+        {
+          type: QuestionType.Number,
+          title: 'On a scale of 1-10, how likely would you be to hire this student?',
+          isRequired: true,
+          key: 'hireabilityNpsRating',
```
Comment (MarkDKoslow) [inline forms/sprint-review.ts]: Can we still restrict these to a range? I think we do for NPS

File: forms/sprint-review.ts [hunk 2]
```diff
@@ -133,5 +133,91 @@ Student can access this feedback through their dashboard`,
...
+        {
+          type: QuestionType.Rating,
+          range: { start: 1, end: 3 },
+          title: 'How did the student demonstrate mastery of the unit content?',
+          isRequired: true,
+          key: 'unitOverallRating',
+        },
+        {
+          type: QuestionType.LongText,
+          title: 'What is blocking the student from passing?',
```
Comment (MarkDKoslow) [inline forms/sprint-review.ts]: More of a product question but this seems like weird wording cause it assumes they didn't pass 🤔. Maybe something like "If the student has not passed the unit, what is blocking them from doing so?" Or we could hide this question unless the previous question is set to a 1

File: forms/sprint-review.ts (hunk 1 above)
Comment (mellogarrett) [reply forms/sprint-review.ts]: Yea good call

File: forms/sprint-review.ts (hunk 2 above)
Comment (mellogarrett) [reply forms/sprint-review.ts]: This is what was on the spec, but I agree it's not worded well. That's a great idea to make it conditional on the question above. I'll reach out to Aparna & Mitchell on this.

File: pages/api/sprint-review.ts [hunk 3]
```diff
@@ -21,8 +22,40 @@ export default async (req: NextApiRequest, res: NextApiResponse) => {
...
+    unitTechnicalRating: data.unitTechnicalRating,
+    unitTeamworkRating: data.unitTeamworkRating,
+    unitRubricItem1Rating: data.unitRubricItem1Rating,
+    unitRubricItem2Rating: data.unitRubricItem2Rating,
+    unitRubricItem3Rating: data.unitRubricItem3Rating,
+    unitRubricItem4Rating: data.unitRubricItem4Rating,
+    unitRubricItem5Rating: data.unitRubricItem5Rating,
+    unitRubricItem6Rating: data.unitRubricItem6Rating,
+    hireabilityNpsRating: data.hireabilityNpsRating,
+  };
```
Comment (MarkDKoslow) [inline pages/api/sprint-review.ts]: Hmm it'd be cool if there were a shorthand way to extract all the keys/values from an object that conform to a type (i.e. the `Keys` types above). I think we could in a loop but not sure that gives us much

Comment (mellogarrett) [reply pages/api/sprint-review.ts]: Yea agreed. I'll try to figure this one out.

Comment (boblauer) [reply pages/api/sprint-review.ts]: You can get pretty close:

```ts
// types/question-keys.ts

export const unitSubmissionReviewProps = [
  'unitOverallRating',
  'blocker',
  'unitTechnicalRating',
  'unitTeamworkRating',
  'unitRubricItem1Rating',
  'unitRubricItem2Rating',
  'unitRubricItem3Rating',
  'unitRubricItem4Rating',
  'unitRubricItem5Rating',
  'unitRubricItem6Rating',
  'hireabilityNpsRating',
] as const;

export type UnitSubmissionReviewKeys = typeof unitSubmissionReviewProps[number];

// api/sprint-review.ts

import { unitSubmissionReviewProps, UnitSubmissionReviewKeys } from '../../types';

async function submitForm(...) {
  const unitReviewSubmission = _.pick(data, unitSubmissionReviewProps);
}
```

Only gotcha is that the `types` file now contains actual javascript and not just types anymore, but being able to define a type and use the array of props from a single array seems like that's worth the trade off.

An alternative would be to define the list of keys in an array in the `api/sprint-review.ts` file, and then define the type within that same file.

```ts
const unitSubmissionReviewKeys = ['a', 'b', 'c'] as const;

async function submitForm(userId: string, data: Record<typeof unitSubmissionReviewKeys>) {
  const unitReviewSubmission = _.pick(data, sprintSubmissionReviewKeys);
}
```

File: pages/api/sprint-review.ts [hunk 4]
```diff
@@ -21,8 +22,40 @@ export default async (req: NextApiRequest, res: NextApiResponse) => {
   }
 };
 
-async function submitForm(userId: string, data: object) {
-  await atlas.instruction.sprintSubmissionReview.submitForm(userId, data);
+async function submitForm(userId: string, data: Record<SprintSubmissionReviewKeys | UnitSubmissionReviewKeys, Answer>) {
```
Comment (boblauer) [inline pages/api/sprint-review.ts]: One thing that stands out here is the types are `XSubmissionReview`, but the data objects are `xReviewSubmission`. Is the transposing of Submissions and Review on purpose?

Comment (mellogarrett) [reply pages/api/sprint-review.ts]: Nope, not on purpose. I'll clean these names up a bit.

File: pages/api/sprint-review.ts (hunk 3 above)
Comment (mellogarrett) [reply pages/api/sprint-review.ts]: Can we use an enum instead of a `type` of pre-defined strings? enums are iterable and live on at runtime.

Comment (mellogarrett) [reply pages/api/sprint-review.ts]: Like 
```ts
export type SprintReviewKeys =
  | 'careerRating'
  | 'careerFeedback'
  | 'interventionReferral'
  | 'otherNotes'
  | 'otherInternal'
  | 'projectManagerId'
  | 'reviewEscalation'
  | 'skillsFeedback'
  | 'skillsRating'
  | 'sprintSubmissionId'
  | 'tags'
  | 'teamworkFeedback'
  | 'teamworkRating'
  | 'threeWordSummary'
  | 'whatWentPoorly'
  | 'whatWentWell';
```
becomes
```ts
export enum SprintReviewKeys {
  careerRating = 'careerRating',
  careerFeedback = 'careerFeedback',
  interventionReferral = 'interventionReferral',
  otherNotes = 'otherNotes',
  otherInternal = 'otherInternal',
  projectManagerId = 'projectManagerId',
  reviewEscalation = 'reviewEscalation',
  skillsFeedback = 'skillsFeedback',
  skillsRating = 'skillsRating',
  sprintSubmissionId = 'sprintSubmissionId',
  tags = 'tags',
  teamworkFeedback = 'teamworkFeedback',
  teamworkRating = 'teamworkRating',
  threeWordSummary = 'threeWordSummary',
  whatWentPoorly = 'whatWentPoorly',
  whatWentWell = 'whatWentWell',
}
```

Comment (mellogarrett) [reply pages/api/sprint-review.ts]: Wow, that `as const` is pretty neat: https://blog.logrocket.com/const-assertions-are-the-killer-new-typescript-feature-b73451f35802/

And I guess the `[number]` just means any index of that const?

Comment (boblauer) [reply pages/api/sprint-review.ts]: Yup, exactly. Got it from here - https://dev.to/shakyshane/2-ways-to-create-a-union-from-an-array-in-typescript-1kd6

**Pull request merged**